sdk/Sdk/Abstractions/SdkRetryOptions.cs

```diff
@@ -5,23 +5,14 @@ namespace Microsoft.Azure.Functions.Worker.Sdk
 {
     internal class SdkRetryOptions
     {
-        public SdkRetryOptions(string strategy, int maxRetryCount, string? delayInterval, string? minimumInterval, string? maximumInterval)
-        {
-            Strategy = strategy;
-            MaxRetryCount = maxRetryCount;
-            DelayInterval = delayInterval;
-            MinimumInterval = minimumInterval;
-            MaximumInterval = maximumInterval;
-        }
+        public string? Strategy { get; set; }
 
-        public string Strategy { get; }
+        public int MaxRetryCount { get; set; }
```

Comment:
My OCD for keeping every possible type as immutable is kicking in.

Reply:
@satvu can you clarify why we're changing this here?

Reply:
For JSON serialization, as far as I understand, STJ expects public properties (that you can get and set) and will only serialize those given the options I see present in `FunctionMetadataJsonWriter`. I also thought immutability was preferable, but I am now understanding why `SdkFunctionMetadata` is also written like this with get and set. Converters are another option, but we already have a pattern established by `SdkFunctionMetadata` and defined SerializationOptions in the writer that work.

Helpful documentation example [here](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to?pivots=dotnet-5-0#serialization-example) to see what gets serialized and what's ignored, and also looked into properties vs fields and how STJ handles them in serialization.